### PR TITLE
Added Casdoor JS SDK

### PR DESCRIPTION
**Pull request by keainye**
Part of: https://github.com/casbin/casnode/issues/256

This is a initial version of Casdoor JS SDK. It only supports password login. 
In next PRs, I will implement these functions:
- OAuth login
- Forget password
- 2FA

Deployed at: https://forum.chromium.link
Username: `Kininaru`
Password: `123`

### CODE REVIEW

**Comment by keainye** [comment]
https://user-images.githubusercontent.com/42638489/125337162-3b4b2980-e381-11eb-9846-482bb7e5d811.mp4

**Comment by hsluoyz** [comment]
@kocoler @oranges-eating @sh1luo  plz review

File: web/src/main/SigninBox.js [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+      problem: [],
+      message: "",
+    };
+
+    let url = `${window.location.protocol}//${window.location.hostname}`;
+    let port = window.location.port;
+    if (port !== "") url = `${url}:${port}`;
+    url += "/callback";
+
+    this.Casdoor = new Casdoor(
```
Comment (kocoler) [inline web/src/main/SigninBox.js]: The `casdoor` instance will be frequently used in various components, would it be better to initialize in `App.js`?

File: web/src/Casdoor/Casdoor.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+export class Casdoor {
+  constructor(endpoint, organization, application, clientId, redirectUri) {
+    this.endpoint = endpoint;
+    this.organizationName = organization;
+    this.applicationName = application;
+    this.clientId = clientId;
+    this.redirectUri = redirectUri;
+
+    fetch(
```
Comment (kocoler) [inline web/src/Casdoor/Casdoor.js]: Asynchronous initialization may be better.

Comment (keainye) [reply web/src/Casdoor/Casdoor.js]: I think `fetch` is already an async function.

File: web/src/main/SigninBox.js (hunk 1 above)
Comment (keainye) [reply web/src/main/SigninBox.js]: That's right, I will move the Casdoor instance to App.js.

File: web/src/Casdoor/Casdoor.js (hunk 2 above)
Comment (kocoler) [reply web/src/Casdoor/Casdoor.js]: All right, it's my wrong.

**Comment by oranges-eating** [comment]
I think it"s ok.

**Review by oranges-eating — APPROVED** [review]
ok

**Comment by hsluoyz** [comment]
@Kininaru 

1. No captcha for login
2. Forget link is broken, at least point to Casdoor's forget page for now.

image

File: web/src/main/SigninBox.js (hunk 1 above)
Comment (keainye) [reply web/src/main/SigninBox.js]: @kocoler Moved to `App.js`

**Comment by keainye** [comment]
@hsluoyz 
1. in old logic, captcha code is sent to Casnode backend with other information, and will be checked in Casnode. But Casdoor does not require captcha, so we can not put captcha here now. We can add it back as soon as Casdoor supports captcha.
2. Pointed to Casdoor temporarily.

**Comment by hsluoyz** [comment]
@Kininaru there's no oauth login.

**Comment by keainye** [comment]
@hsluoyz
Supported OAuth. Also deployed at: https://forum.chromium.link

**Comment by hsluoyz** [comment]
@Kininaru can we at least restore our old looking? This one is way too "rough".

image

You can get our old looking from previous commits:

image

**Comment by keainye** [comment]
@hsluoyz Finished, now it looks like it used to be. https://forum.chromium.link

image

**Comment by hsluoyz** [comment]
@Kininaru great! Next step would be sign up page, user profile page, etc. Don't hesitate to add new APIs to Casfoor if you need more data to build the pages.

**Pull request merged**